mobile/library/java/io/envoyproxy/envoymobile/engine/EnvoyEngine.java

```diff
@@ -22,22 +22,31 @@ public interface EnvoyEngine {
    */
   void terminate();
 
+  /**
+   * Performs any registrations necessary before running Envoy.
+   *
+   * The envoyConfiguration is used to determined what to register.
+   *
+   * @param envoyConfiguration The EnvoyConfiguration used to start Envoy.
+   */
+  void performRegistration(EnvoyConfiguration envoyConfiguration);
+
   /**
    * Run the Envoy engine with the provided yaml string and log level.
    *
-   * The envoyConfiguration is used to resolve the configurationYAML.
+   * This does not perform registration, and performRegistration may need to be called first.
```

Comment:
nit: consider changing performRegistration to `performRegistration()`

Reply:
done